controllers/openstackcluster_controller.go

```diff
@@ -215,10 +215,26 @@ func deleteBastion(scope scope.Scope, cluster *clusterv1.Cluster, openStackClust
 		return err
 	}
 
-	instanceName := fmt.Sprintf("%s-bastion", cluster.Name)
-	instanceStatus, err := computeService.GetInstanceStatusByName(openStackCluster, instanceName)
-	if err != nil {
-		return err
+	if openStackCluster.Status.Bastion != nil && openStackCluster.Status.Bastion.FloatingIP != "" {
+		// Floating IP could have been created but not associated to instance, attempt to delete it from saved status first
```

Comment:
Actually this is the regular case (floating IP is stored in status, and deleted here), I haven't removed [the other block](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/d2070463c2d17d81b44b0f3fe43eba4cd1910553/controllers/openstackcluster_controller.go#L241-L254) that removes floating IP attached to bastion, but it will we be useful only if floating IP is was not properly saved in status for some reason. Maybe I should reword the comment to make it clearer.

Reply:
As the removal of floating IP will already be logged [when it will be removed](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/8a49b9f23c3a8bd2f41c4e071213edc8bc8ede7b/pkg/cloud/services/networking/floatingip.go#L111) at this point, it does not seem very meaningfull to add another log. I [changed my comment](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/19a6cb5fff75a67b792cf274f37d55b1738c93e1/controllers/openstackcluster_controller.go#L248), hoping that it will be more explicit. Let me know if you want me to revisit that.